iothub/service/src/Authentication/AuthenticationMethodFactory.cs

```diff
@@ -12,17 +12,6 @@ public sealed class AuthenticationMethodFactory
     {
         internal static IAuthenticationMethod GetAuthenticationMethod(IotHubConnectionStringBuilder iotHubConnectionStringBuilder)
         {
-            if (!string.IsNullOrWhiteSpace(iotHubConnectionStringBuilder.DeviceId))
-            {
-                if (!string.IsNullOrWhiteSpace(iotHubConnectionStringBuilder.SharedAccessKey))
-                {
-                    return new ServiceAuthenticationWithDeviceSharedAccessPolicyKey(iotHubConnectionStringBuilder.DeviceId, iotHubConnectionStringBuilder.SharedAccessKey);
-                }
-                if (!string.IsNullOrWhiteSpace(iotHubConnectionStringBuilder.SharedAccessSignature))
-                {
-                    return new ServiceAuthenticationWithDeviceSharedAccessPolicyToken(iotHubConnectionStringBuilder.DeviceId, iotHubConnectionStringBuilder.SharedAccessSignature);
-                }
-            }
             if (string.IsNullOrWhiteSpace(iotHubConnectionStringBuilder.SharedAccessKey))
             {
                 return new ServiceAuthenticationWithSharedAccessPolicyToken(iotHubConnectionStringBuilder.SharedAccessKeyName, iotHubConnectionStringBuilder.SharedAccessSignature);
```

Comment:
Were you able to check if ServiceAuthenticationWithSharedAccessPolicyToken is relevant for service clients? Or, are you planning to look into that as a part of a separate task?

Reply:
Its being asserted in unit tests [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/e9511059a5a02126fd7d4354deb2919a27c76ee6/iothub/service/tests/ConnectionString/ConnectionStringTests.cs#L63)
It seemed relevant but digging more into it, looks like its not very useful.
I'll remove it